doc/md/versioned/troubleshoot.mdx

```diff
@@ -0,0 +1,578 @@
+
+:::
+
+<Tabs
+defaultValue="apply_stdout"
+values={[
+{label: 'atlas migrate apply (STDOUT)', value: 'apply_stdout'},
+{label: 'atlas migrate apply (STDERR)', value: 'apply_stderr'},
+{label: 'atlas migrate status', value: 'status'},
+]}>
+<TabItem value="apply_stdout">
+
+```text
+$ atlas --env local migrate apply 1 --tx-mode none
+Migrating to version 2 from 1 (1 migrations in total):
+
+  -- migrating version 2
+    -> INSERT INTO `users` (`id`, `name`) VALUES (1, 'masseelch'), (2, 'rotemtam'), (3, 'a8m');
+    -> INSERT INTO `groups` (`id`, `name`) VALUES (1, 'Founders'), (2, 'Senior Engineers'), (3, 'Junior Engineers');
+    -> INSERT INTO `user_groups` (`user_id`, `group_id`) VALUES
+    -- Founders
+    (2, 1), (3, 1),
+    -- Seniors
+    (1, 2), (2, 2), (3, 2),
+    -- Constraint error (adding masseelch twice to seniors)
+    (1, 2);
+// highlight-next-line-error-message
+     Error 1062: Duplicate entry '1-2' for key 'user_groups.PRIMARY'
+
+  -------------------------
+  -- 2.521007ms
+  -- 0 migrations ok (1 with errors)
+  -- 2 sql statements ok (1 with errors)
+
+// highlight-next-line-error-message
+Error: Execution had errors:
+// highlight-next-line-error-message
+Error 1062: Duplicate entry '1-2' for key 'user_groups.PRIMARY'
+```
+
+</TabItem>
+<TabItem value="apply_stderr">
+
+```text
+$ atlas --env local migrate apply 1 --tx-mode none
+Error: sql/migrate: execute: executing statement "INSERT INTO `user_groups` (`user_id`, `group_id`) VALUES
+    -- Founders\n    (2, 1), (3, 1),
+    -- Seniors\n    (1, 2), (2, 2), (3, 2),
+    -- Constraint error (adding masseelch twice to seniors)
+    (1, 2);" from version "2": Error 1062: Duplicate entry '1-2' for key 'user_groups.PRIMARY'
+```
+
+</TabItem>
+<TabItem value="status">
+
+```text
+$ atlas --env local migrate status
+Migration Status: PENDING
+  -- Current Version: 2 (2 statements applied)
+  -- Next Version:    2 (1 statements left)
+  -- Executed Files:  2 (last one partially)
+  -- Pending Files:   2
+
+Last migration attempt had errors:
+  -- SQL:   INSERT INTO `user_groups` (`user_id`, `group_id`) VALUES     -- Founders     (2, 1), (3, 1),     -- Seniors     (1, 2), (2, 2), (3, 2),     -- Constraint error (adding masseelch twice to seniors)     (1, 2);
+// highlight-next-line-error-message
+  -- ERROR: Error 1062: Duplicate entry '1-2' for key 'user_groups.PRIMARY'
+```
+
+</TabItem>
+</Tabs>
+
+## Fixing Migration Failures
+
+In our experience, the most common failures are introduced by one of the following issues:
+
+- **Syntax Error**: One SQL statement contained a syntax error. Since Atlas provides a way to [lint](/versioned/lint) a
+migration directory, this case happens due to a misconfiguration, e.g. not executing the linter before applying the
+migration or applying against the wrong dialect (like SQLite files against MySQL).
+- **DDL Semantics**: One SQL statement was semantically incorrect, e.g. adding a table that does already exist. This
```

Comment:
The term "Semantics" means something else to me. Like compiler sematic-checks, which is the "meaning of the DDL" in this case, and not its result. I would change it to "Schema Dependant Change".